assets/app/scripts/app.js

```diff
@@ -17,24 +18,121 @@ angular
     'ngSanitize',
     'ngTouch'
   ])
-  .config(function ($routeProvider, $locationProvider) {
+  // configure our tabs and routing
+  .config(['$routeProvider', 'HawtioNavBuilderProvider', function($routeProvider, builder) {
+    var template = function() {
+      // TODO - Don't love triggering the show/hide drawer here, would prefer if 
+      // we could listen for an event that the nav was being redrawn and 
+      // check HawtioNav.selected()
+      if (this.isSelected()) {
+        if (this.tabs && this.tabs.length > 0) {
+          $("body").addClass("show-drawer");
+        }
+        else {
+          $("body").removeClass("show-drawer");
+        }
+      }
+      return "<sidebar-nav-item></sidebar-nav-item>";        
+    };
+
+    var projectHref = function(path) {
+      return function() {
+        var injector = HawtioCore.injector;
+        if (injector) {
+          var routeParams = injector.get("$routeParams");
+          if (routeParams.project) {
+            return "/project/" + routeParams.project + "/" + path;
+          }
+        }
+        return "/project/:project/" + path; 
+      }
+    };
+
+    // TODO - hawtio nav runs the href() function once during initial
+    // build and then uses that in the isSelected function it creates
+    // we need to always get the current href which is dependent on 
+    // the current project.  Want to propose upstream fix to always
+    // run item.href()
+    var isSelected = function() {
+      var injector = HawtioCore.injector;
+      if (injector) {
+        var location = injector.get("$location");
+      }
+      return location.path().indexOf(this.href()) === 0;
```

Comment:
good call!  Will push an update in a couple minutes...

On Mon, Jan 19, 2015 at 4:30 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In assets/app/scripts/app.js
> <https://github.com/openshift/origin/pull/659#discussion-diff-23188128>:
>
> > +        }
> > +        return "/project/:project/" + path;
> > +      }
> > +    };
> > +
> > +    // TODO - hawtio nav runs the href() function once during initial
> > +    // build and then uses that in the isSelected function it creates
> > +    // we need to always get the current href which is dependent on
> > +    // the current project.  Want to propose upstream fix to always
> > +    // run item.href()
> > +    var isSelected = function() {
> > +      var injector = HawtioCore.injector;
> > +      if (injector) {
> > +        var location = injector.get("$location");
> > +      }
> > +      return location.path().indexOf(this.href()) === 0;
>
> Doesn't have to be an exact match, just make a partial match require a
> trailing '/'.
> return (location.path() == this.href()) ||
> (location.path().indexOf(this.href() + "/") === 0);
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/659/files#r23188128>.
>